esp-hal-common/src/utils.rs

```diff
@@ -0,0 +1,232 @@
+//! Utils
+//! 
+//! # Overview
+//! 
+//! Collection of struct which helps you write to registers.
+
+use crate::peripherals::generic::{Reg, RegisterSpec, Resettable, Writable, Readable};
+
+const U32_ALIGN_SIZE: usize = core::mem::size_of::<u32>();
+
+const U32_FROM_BYTES: fn([u8; 4]) -> u32 = u32::from_ne_bytes;
+const U32_TO_BYTES: fn(u32) -> [u8; 4] = u32::to_ne_bytes;
+
+// The alignment helper helps you write to registers that only accepts u32
+// using regular u8s (bytes). It keeps a write buffer of 4 u8 (could in theory
+// be 3 but less convient). And if the incoming data is not convertable to u32
+// (i.e. not a multiple of 4 in length) it will store the remainder in the
+// buffer until the next call.
+//
+// It assumes incoming `dst` are aligned to desired layout (in future
+// ptr.is_aligned can be used). It also assumes that writes are done in FIFO
+// order.
+#[derive(Debug)]
+pub(crate) struct AlignmentHelper {
```

Comment:
I believe this comes from `sha.rs`, can we remove it from there if it exists here?

Reply:
OK and I will adapt `sha.rs` for calling it.